artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTConnection.java

```diff
@@ -216,8 +220,10 @@ public void flush() {
    public void bufferReceived(Object connectionID, ActiveMQBuffer buffer) {
    }
 
-   public void setConnected(boolean connected) {
-      this.connected = connected;
+   @Override
+   public void connected() throws Exception {
+      remotingService.sendConnectedNotification(this);
```

Comment:
This should be responsibility of core system to sending notification so connected notification is generic.

Reply:
see other comments, ive dug deeper on this and seems actually the act of mqtt connected == core server session created.